test/test.js

```diff
@@ -83,6 +106,58 @@ describe('EnterpriseSearchClient', () => {
       ])
     })
   })
+
+  context('#permissions', () => {
+    it('should list permissions', async () => {
+      const results = await client.getPermissions(mockContentSourceKey)
+      assert.deepEqual(results, {
+        meta: {
+          page: { current: 1, total_pages: 1, total_results: 2, size: 25 }
+        },
+        results: [
+          { user: 'elastic', permissions: [] },
+          { user: 'enterprise_search', permissions: [] }
+        ]
+      })
+    })
+
+    it('should pass page size', async () => {
```

Comment:
Do we need all 3 of these tests? size, current, and current + size?

Reply:
Can we just have one that passes & size?

Reply:
That's a habit I got from another place, where we had an automatic test coverage check that blocked PRs that didn't cover every logic branch. It was excessive sometimes, but here I find the tests useful.

The test that checks for only "current: 2" shows that API sends a response with an empty array in results, and not 
a) some kind of error
or
b) automatically changes paging params to default (current:1) or shows the last page that has results.
(But I will rename it to "accept paging parameters outside of results range")

The test that checks for only "size: 1" confirms that API accepts a request with only that parameter. I agree that additional value of this test is pretty small but I think that it needs a very small investment and having a little higher coverage for this investment is a net positive. 

If you still feel strong about it, I'm fine with deleting this test.

I'm trying to be practical with this approach though. If I had 8 conditionals, I wouldn't cover all possible combinations of them.

Does it make sense?

Reply:
> The test that checks for only "size: 1" confirms that API accepts a request with only that parameter.

That's testing the API though, right? To test this client, we just need to verify that the parameters are passed to the API correctly, and that different variations in responses are handled correctly.

I also do not have a strong opinion about this, but I think the tests might end up getting excessive if we test like this.

Again, really just exercising our code, not the API code. As long as a parameter is passed along then it's 👌in my opinion.

I'm going to approve this PR.

Reply:
> That's testing the API though, right?
I was thinking it myself but then I thought that if we wanted to only unit test our clients, we could just check that function returns the correct unresolved request Promise. But since we manually record API responses, that makes it more of an integration test (although with mocked response) and it's ok to have different params combinations for integration tests.